### PR TITLE
DOC: add sdg-2023-r1 proposal

### PR DESCRIPTION
**Pull request by jorgepiloto**
As per the incomming [Small Development Grants](https://numfocus.org/programs/small-development-grants) cycle, I would like to submit this proposal whose goal is to enhance the visualization capabilities of poliastro. This goal is accomplished by:

* Adding a powerful and customizable PyVista backend
* Implementing animation capabilities
* Explore new miscellaneous options like frame plotting or celestial sphere support

### CODE REVIEW

File: numfocus-sdg-2023-r1.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  benefit without requiring to install poliastro.
+  
+* Since PyVista is an affiliated NumFOCUS project, all the community can benefit
+  from the grant in one way or another.
+
+* The testing of the new backend can be performed using the pytest-pyvista
+  plugin for pytest. The adoption of this plugin in poliastro can generate
+  useful feedback for enhancing the plugin.
+
+All the proposed features would allow poliastro to be closer to commercial
```
Comment (astrojuanlu) [inline numfocus-sdg-2023-r1.md:91]: :100:

File: numfocus-sdg-2023-r1.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+------------------------
+
+The project, as depicted in the "Timeline" section of this proposal, is expected
+to be completed within the course of 5 months. Starting date would be May 2023
+and the project would be completed by the end of September 2023.
+
+The grant is used as hourly rates and number of work hours for contractors. The
+requested budget is justified as follows:
+
+4500$ = 5 [month] * 15 [day / month] *  3 * [hour / day] * 20.00 [$ / hour]
```
Comment (astrojuanlu) [inline numfocus-sdg-2023-r1.md:112]: So, more or less 12 hours a week, right?

**Review by astrojuanlu — APPROVED** [review]
I think this is super promising! Approved with some style comments and a mention to Cesium

File: numfocus-sdg-2023-r1.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+------------------------
+
+Implement a powerful and customizable visualization backend based on PyVista and
+provide plotting animation capabilities to poliastro.
+
+
+6 - Description of the proposal
+-------------------------------
+
+poliastro is an open source (MIT) collection of Python functions useful in
```
[the remark this answers is not in the document]
Comment (jorgepiloto) [reply numfocus-sdg-2023-r1.md]: You are right. Let us be consistent with the current definition from the documentation.

File: numfocus-sdg-2023-r1.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+Astrodynamics and Orbital Mechanics, focusing on interplanetary applications. It
+provides a simple and intuitive API and handles physical quantities with units.
+
+Visualizing astronomical objects together with their path and motion is a key
+feature of any Astrodynamics library, including poliastro.
+
+At the moment, poliastro supports two backends built on top of the Matplotlib
+library and the Plotly library. However, these libraries are not conceived for
```
[the remark this answers is not in the document]
Comment (jorgepiloto) [reply numfocus-sdg-2023-r1.md:47]: True. Let me rephrase this paragram so it becomes clear that althought the libraries support 3D visualization, their capabilities are limited.

[the remark this answers is not in the document]
Comment (jorgepiloto) [reply numfocus-sdg-2023-r1.md:47]: Regarding Cesium, I will make the following points:

* As of today, it only supports Earth as the attractor.
* It requires to use a browser or Jupyter to access the visualization.

On the other hand:

* Since PyVista is a generic graphics library, all sorts of attractors are supported at the cost of programming those by our means.
* PyVista runs in dedicated windows, notebooks or even in the web.

File: numfocus-sdg-2023-r1.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+three-dimensional visualization and become a limitation when requiring complex
+visualization techniques like using textures.
+
+A solution proposed during the SciPy 2022 conference was to create a new backend
+based on PyVista. This library provides a Pythonic high-level API to The
+Visualization Toolkit (VTK) library.
+
+Another desired feature in poliastro is animating plots. This feature becomes
```
[the remark this answers is not in the document]
Comment (jorgepiloto) [reply numfocus-sdg-2023-r1.md:62]: Good catch!

File: numfocus-sdg-2023-r1.md (hunk 2 above)
Comment (jorgepiloto) [reply numfocus-sdg-2023-r1.md]: That's it. I think this is a reasonable amount of hours that I can accomplish over the course of 5 months.
